src/librustc_driver/driver.rs

```diff
@@ -592,7 +597,7 @@ pub fn phase_2_configure_and_expand<'a, F>(sess: &Session,
                                    addl_plugins.take().unwrap())
     });
 
-    let mut registry = Registry::new(sess, &krate);
+    registry.krate_span = krate.span;
```

Comment:
I found that from line https://github.com/rust-lang/rust/blob/master/src/librustc_driver/driver.rs#L567 ,
'krate' is updated during the compilation process.

If krate.span is not affected through L567 to L597, I agree that this is unnecessary process.